getinsomnia/insomnia
# feat: add reset passphrase [INS-3206]

### Description
Adding the support for reset passphrase. When resetting the encryption passphrase, we are nullifying the encrypted content records in the remote (as users are the only ones to decrypt and no longer can decrypt it without knowing the passphrase).

This means we need to reconcile any local data that may still exist.
1. relinking these local files to the remote project
2. re-versioning these files and syncing to the cloud (that the users have opted in previously)
3. clean up the remote book-keeping record

The backend book-keeps the deleted file ids when resetting the passphrase and the electron app can know the account has reset the passphrase. It will nullify the existing session as the session data has to do with the encryption key, and the user is asked to re-sign in with the new passphrase (this is the case when user has chosen to reset passphrase during the sending invite flow, but still signed into the electron app => invitation requires passphrase for project encryption. Therefore, the encryption keys must be reconciled with the new keys).

The user will be redirected to the data recovery page where database operation and versioning operation run. During this operation, the remote will also clean up these records, so the user won't be in the state of reset passphrase flag all the time. The electron app will clear the session storage, and move onto the organization page.

### Steps to test
1. sign into your account in staging
2. create 3 team projects with cloud option
3. create files in each of them
4. invite a member
5. go to the forgot passphrase page and reset your passphrase
6. re-sign into the app
7. confirm all the files were where they were
